samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -423,9 +428,61 @@ static bool scan_check_and_sync_broadcast(struct bt_data *data, void *user_data)
 	return false;
 }
 
+static bool is_substring(const char *substr, const char *str)
+{
+	const size_t str_len = strlen(str);
+	const size_t sub_str_len = strlen(substr);
+
+	if (sub_str_len > str_len) {
+		return false;
+	}
+
+	for (size_t pos = 0; pos < str_len; pos++) {
+		if (tolower(substr[pos]) == tolower(str[pos])) {
+			if (pos + sub_str_len > str_len) {
+				return false;
+			}
+
+			if (strncasecmp(substr, &str[pos], sub_str_len) == 0) {
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+
+static bool data_cb(struct bt_data *data, void *user_data)
+{
+	char *name = user_data;
+
+	switch (data->type) {
+	case BT_DATA_NAME_SHORTENED:
+	case BT_DATA_NAME_COMPLETE:
+	case BT_DATA_BROADCAST_NAME:
+		memcpy(name, data->data, MIN(data->data_len, NAME_LEN - 1));
+		return false;
+	default:
+		return true;
+	}
+}
+
+
 static void broadcast_scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_simple *ad)
 {
+	struct net_buf_simple buf_copy;
+	char name[NAME_LEN] = {0};
+
 	if (info->interval != 0U) {
+		/* call to bt_data_parse consumes netbufs so shallow clone for verbose output */
+		if (strlen(CONFIG_TARGET_BROADCAST_NAME) > 0U) {
+			net_buf_simple_clone(ad, &buf_copy);
```

Comment:
```suggestion

	if (info->interval != 0U) {
		/* call to bt_data_parse consumes netbufs so shallow clone for verbose output */
		if (strlen(CONFIG_TARGET_BROADCAST_NAME) > 0U) {
			struct net_buf_simple buf_copy;
			char name[NAME_LEN] = {0};
	
			net_buf_simple_clone(ad, &buf_copy);
```

They can moved to reduce the scope